### DISCUSSION TITLE
Multiple Platforms requiring the same MEDIUM ID

### DISCUSSION THREAD

**Discussion by jbatterman** [opendcs/opendcs#185] [Ideas]
Our USACE office has been exploring with different web pulls through OpenDCS, and discovered a potential issue.  So far we've been able to generate a workaround, but that may not always be possible.

For each abstractweb source, it's a unique URL with a $MEDIUMID that uses the MEDIUM ID in the corresponding Netlist platforms to pull data.  We've found that some web sources require different abstract URLs for different types of data (i.e., daily vs. instantaneous).  In these cases, the MEDIUM ID is the same, but the platform needs to be different, since it uses a different config to decode the data.  We can have multiple platforms connected to one site by using the Designator box in the GUI, but are not able to connect different platforms to one MEDIUM ID, even if they have different Designators.  Our workaround so far is to modify the MEDIUM ID to contain part of the static version of the URL as well, so that they look different.  In one case, using "BBR+" for one and "BBR" for the other, since the web URL uses + signs to fill in spaces.

I don't have a lot of background knowledge of OpenDCS, but in my limited knowledge, I'd like to propose the idea of allowing multiple platforms to have the same MEDIUM ID if they have different designators, similar to the Site.

Another USACE office ran into this issue with a different abstractweb source as well.  I've attached the email conversation as a text file below if you'd like more background or explanation of the issue.  We look forward to hearing what other ideas might solve this problem, or if our idea is implementable in the future.  Thanks!

[emails.txt](https://github.com/opendcs/opendcs/files/10807877/emails.txt)

**Comment by jbatterman** [comment]
Additional thought I had while marinating on this problem.  We've noticed our Platforms tab is getting pretty cluttered with all our web pulls we are adding.  Each type of data pull requires its own Platform right now.  Could another possibility be a way to have paired selectors, so you connect one Site, then have a Config/Transport Media selector where you can choose as many pairs as you want?  So for example, have a config called "USGS-DailyStage" and a Transport Media with a MEDIUM ID of the USGS ID.  Then could have a "USGS-HourlyStage" and a Transport Media with the same ID (or different for those cases).  And maybe a third with a config called "GOES" and a Transport Media with a MEDIUM ID of the DCP ID.  Again, unsure on the amount of work involved in this, but that sounds like it could clear up the clutter of the Platform tab.

**Comment by agilmore2** [comment]
It occurs to me that we have different decoding scripts for self-timed vs
random transmissions. If there's a way to tie decoding scripts to web URLs
etc., that would seem to fit.

**Comment by MikeNeilson** [comment]
It seems like the in this context the "MEDIUM" is really just shorthand for "fill out the weblink to get the right data."

Perhaps it would make more sense to have a field in the Platform edit that was "real station id-> config" or "config -> real station id".

I could see adding a self reference parent field to handle that so they wouldn't show up in the main list (other then maybe a count.)

@jbatterman is that a reasonably fair assessment of how the primary MEDIUM id gets used to pull the report. Seemed like the report you provided contained several different stations and intervals/datatypes under the one report. Still not entirely sure how that would work out with the decodes script, but I'm starting to think that isn't step one.

**Comment by jbatterman** [comment]
Yes, I think that covers how MEDIUM ID is utilized.  For the abstract web pull, we substitute $MEDIUMID in the URL with the Transport Media Medium ID from the Platform tab.  And for GOES DCP data pulls, the Transport Media Medium ID along with the channel, transport time, etc. is used to look for DCP messages.  So certain data pulls do need more info than just the Medium ID.  But if there's a way we can connect those to the different format statements from a config into one Platform it would clean up that tab quite a bit.  And remove the need for Designators, too, at least in this use case.
